### ISSUE TITLE
Bug: App crashes when previewing large files

### ISSUE THREAD

**Issue by Marin-Kitagawa** [files-community/Files#12151]
### Description

Files often resets itself saying that it ran into a problem.  All the tags gets reset and all custom tags are deleted. All layouts are changed. 

![image](https://user-images.githubusercontent.com/49131888/233762049-c6357f2d-d527-47d6-ac64-f8d6743affcb.png)

![image](https://user-images.githubusercontent.com/49131888/233762751-9db968da-cd35-4903-babb-bc53661676fa.png)

### Steps To Reproduce

1. Open files and wait
2. It will close itself and reset everything

### Requirements

- Fix the crash and restore the tags and layouts including custom tags
- 

### Files Version

Current version: 2.4.66.0 [Stable]
Previously I was using Preview version of 2.4.66. Then, I uninstalled it and installed the stable version from Microsoft Store. So, the above problem is in the stable version of 2.4.66

Now, I installed 2.4.67 using the classic installer as Microsoft Store only has 2.4.66 version. Even in this I get the same issue. When I close and open the app again, all the tags are gone

### Windows Version

Windows 11 Pro 10.0.22621.1555

### Log file

[debug.log](https://github.com/files-community/Files/files/11300260/debug.log)

**Comment by hishitetsu** [comment]
Thank you for your feedback. The following log was output, so it appears that there is not enough memory to run the application stably.

`System.OutOfMemoryException: Insufficient memory to continue the execution of the program.`

**Comment by Marin-Kitagawa** [comment]
I have $16$ `GiB` of RAM with nothing running other than Files (this software), Notepad++, Teams, File Explorer and `C:\` drive has over $60$ `GiB` of free space. These programs doesn't even consume more than $8$ `GiB` of RAM. So, why does this occur?

![image](https://user-images.githubusercontent.com/49131888/233763707-798f47d9-9142-44e4-9d34-67013b61851e.png)

**Comment by hishitetsu** [comment]
It seems to occur when trying to display a preview of an audio or movie file. Is it possible that you are previewing a very large or corrupt file?

> Please push the latest version to Microsoft Store.

I believe the timing of delivery is controlled by Microsoft.

**Comment by Marin-Kitagawa** [comment]
Yeah, I was previewing a file of size $206$ `MiB`. Is this considered very large? Is it possible to remove that limitation or add some safeguards which says "File too large to preview" (something of this sort)?

> I believe the timing of delivery is controlled by Microsoft.

Oh ok, then

**Comment by hishitetsu** [comment]
I tried it with an MP4 movie that is over 1GB, but I can't reproduce it. Can you tell me which type of file causes the crash?

**Comment by Marin-Kitagawa** [comment]
> I tried it with an MP4 movie that is over 1GB, but I can't reproduce it. Can you tell me which type of file causes the crash?

It is a `MP4` file

**Comment by Marin-Kitagawa** [comment]
> I tried it with an MP4 movie that is over 1GB, but I can't reproduce it. Can you tell me which type of file causes the crash?

It is an `MP4` file

**Comment by yaira2** [comment]
@Marin-Kitagawa can you confirm if this is fixed in v2.4.70?

**Comment by Marin-Kitagawa** [comment]
@yair100 The new version is not found in MS Store

![image](https://user-images.githubusercontent.com/49131888/235281131-9cbc80d4-3a2b-43e9-ae6a-d29c5143b1c4.png)

Also, it is $2.4.69$ instead of $2.4.79$ right?

If yes, then the latest version $2.4.70$ is yet to become available

**Comment by yaira2** [comment]
@Marin-Kitagawa the changelog in the Store had the wrong number (typo), it has since been updated to have the correct version number v2.4.70.

**Comment by Marin-Kitagawa** [comment]
@yair100 Yes, it still happens in $2.4.70$ as well.

This is what I did
1. Enabled tags widget. I always do this
2. Changed the layout to `Columns`
3. Changed the default sorting to `Date Modified` and order to Descending
4. Enabled the preview pane
5. Clicked on an `MP4` file of size $104.82$ `MiB` and it crashed

**Comment by yaira2** [comment]
@Marin-Kitagawa is this similar to #11936?

**Comment by Marin-Kitagawa** [comment]
I am not sure of the codecs that the files I preview uses though. I only know about the file extensions. I have given the log files above. According to the issue you mentioned, the app doesn't store the logs files. But @hishitetsu was able to find that I was previewing some media which was the case. Also, it restarts itself just after the crash with everything reset

**Comment by Marin-Kitagawa** [comment]
And one more thing is that, at first it didn't crash. For all the time I opened it after previewing videos of such large size, it didn't crash. But when I close and opened for the first time after successfully previewing such large video file, the tags were reset. I again changed the name of the tags and previewed the video files. Now, only one tag got reset (changed to `Unknown`; before both the tags (as in the above (first) image)) was changed to `Unknown).  On third and subsequent tries, nothing changed; no reset occurred.
